File: app/src/main/java/com/example/dcct/model/QueryModel.java
package com.example.dcct.model;

import com.example.dcct.bean.BackResultData;
import com.example.dcct.bean.PostQueryEntity;
import com.example.dcct.bean.QueryResultEntity;

import java.util.List;

public interface QueryModel{
    void postQueryInformation(PostQueryEntity postQueryEntity,ReturnQueryResult returnQueryResult);
    interface ReturnQueryResult{
        void onComplete(BackResultData<List<QueryResultEntity>> listBackResultData);
        void onError(String msg);
    }
}

File: app/src/main/java/com/example/dcct/model/QueryModel.kt
package com.example.dcct.model

import com.example.dcct.bean.BackResultData
import com.example.dcct.bean.PostQueryEntity
import com.example.dcct.bean.QueryResultEntity

interface QueryModel {
    fun postQueryInformation(postQueryEntity: PostQueryEntity, returnQueryResult: ReturnQueryResult)
    interface ReturnQueryResult {
        fun onComplete(listBackResultData: BackResultData<List<QueryResultEntity>>)
        fun onError(msg: String)
    }
}